### ISSUE TITLE
ES7 - funder names not always using latest

### ISSUE THREAD

**Issue by robredpath** [ThreeSixtyGiving/grantnav#512]
See also #409 

>  The National Lottery Community Fund is showing under it's old name The Big Lottery Fund

**Comment by robredpath** [comment]
See also #510

**Comment by robredpath** [comment]
I just had a call with @michaelwood about this, and wanted to record some notes here so that everyone knows what's happening! 

Sometimes, funders change their names. When they do so, they often start publishing new data using the new name, but keep the old data using their old name. So, if ABCD Charity (GB-CHC-11112222) becomes WXYZ Charity (GB-CHC-11112222) on the 1st of Jan 2020, then all pre-2020 grants would be recorded as being from ABCD Charity, while all 2020 and later grants would be from WXYZ Charity. Some publishers split files by date, some don't - so the change in name may occur somewhere within a file. Some publishers may choose to change all their old data to use the new name - that is, of course, their prerogative and doesn't cause us any technical issues. 

There are five places where funders are named - the **[Funders](https://grantnav.threesixtygiving.org/funders) page**, the **individual Funder page** (eg [Greater London Authority](https://grantnav.threesixtygiving.org/funder/GB-LAE-GLA)) on the **individual grant result** in search results, in the left-hand **Filter** on the search results page, and on the **individual grant page**.

As per https://github.com/ThreeSixtyGiving/grantnav/issues/409#issuecomment-346324847 and https://github.com/ThreeSixtyGiving/grantnav/issues/409#issuecomment-346590894 , the **Funders page**, the **individual funder page** and the **Filter** use the `Funding Org: Name` for the first grant in the first file that has a `Funding Org: ID` that's come across when loading. The **individual grant page** and the **individual grant result** use the contents of the individual grant. 

As part of switching GrantNav's data source to the data store rather than datagetter, we should check in on whether this is still what we want to do, and if it still makes sense, now that the concept of 'the first file' is a bit more removed from GrantNav itself. 

@michaelwood - is it possible to duplicate the current functionality using the data store? Is there any mileage in making funder name -> funder id a many-to-one mapping, or is that just a recipe for disaster and UI pain?

@KDuerden @drkane - are you happy with the current way that GrantNav handles this? If you had complete freedom, is there some other way you'd prefer this to be handled?

**Comment by drkane** [comment]
My comment from the plan.io thread:

> > The National Lottery Community Fund is showing under it's old name The Big Lottery Fund (can't remember how this has been changed in the current one)
> 
> I was actually wrong - this same behaviour is currently happening on current and new versions of GrantNav, so this isn't a specific issue relating to the new one. I think David R is correct that we do need to be careful about the file ordering though - as I understand it GrantNav relies on the file ordering to find the canonical name for a funder (and uses the name from the first row of the first file, which the registry ensures is the latest one).
> 
> We can probably do something more systematic for the new grantnav upload process, and maybe order the grants by funder then by award date so it ensures the latest funder name from the latest grant is used.

Given the flexibility of the new datastore I think something like the ordering above would make sense.

NB I think you've missed one place where the funder name appears - it's not technically the funder name but for the user I think the publisher name that appears on individual grant records and in other places would also be the funder name 90% of the time. This confused me - the Big Lottery Fund is shown as TNCF as the publisher name.

Note that there's the same problem for recipient organisations - except that the solution is different. And it's arguably a bigger problem. I think for those it might be best to use the name given on the biggest grant, or the name with the most grants.

**Comment by michaelwood** [comment]
There is also an issue here about when/where data manipulation is actually done. If I were a user and downloaded the dataset from grantnav or the datastore I think I would expect to see the data as you would see it in the UI. 
Which means we would need to change the name at the datastore level rather than the grantnav level?

This could be done at the time of loading the data into the datastore from the datagetter and would therefore change all the data from that point onwards or we could change the names at the point of generating the "latest" data. 

Either way I would recommend not to rely on any implicitness from the ordering to do this, a mapping table I think would be best for this on the datastore itself, this would be flexible and easy to add future mappings.

**Comment by KDuerden** [comment]
> If I were a user and downloaded the dataset from grantnav or the datastore I think I would expect to see the data as you would see it in the UI.
None of this previously has been about changing or manipulating the data. It's only been about how names fields get populated in certain bits of GrantNav's UI - with none of the underlying data being changed. Moving to the datastore method of populating GrantNav shouldn't be changing the principle that the published data is the authoritative source of information, which might get enriched but wouldn't have names changed.

**Comment by drkane** [comment]
I'd definitely agree that this is a GrantNav issue rather than datastore at the moment. 

It could be something we look at for the datastore, but we'd need to think carefully about the messaging around it if we were altering the originally published data in what people can download from the datastore.

**Comment by drkane** [comment]
A further complication on this - I think the GrantNav filters use a derived field called `id_and_name` - I'm not sure how this field is created but I guess this could result in a situation where the funder ends up appearing twice in the funder filter, which wouldn't be the desired behaviour.

**Comment by michaelwood** [comment]
Proposed solution for both es7 and current master https://github.com/ThreeSixtyGiving/grantnav/issues/519

**Comment by robredpath** [comment]
#681 probably fixes this; would still be useful to check across all places identified ^^ before closing

**Comment by robredpath** [comment]
@mariongalley that's a useful example - thanks. As ever, it's that way because we're presenting the information from the actual grant record published by the publisher. We could use the canonical name in the search results template, I think. 

So, let's leave this open until we get to fixing that!

**Comment by KDuerden** [comment]
This would be a move away from being a mirror to the data, and it's worth thinking through what it would mean to provide a name in search results but different one in the grant record (unless we're proposing to overwrite what is given in each record too?).

**Comment by robredpath** [comment]
> This would be a move away from being a mirror to the data

Yes, although I'm not sure how far the right-hand side of results cards is a mirror to the data as it stands: we already display location data that's derived, without any visual differentiation. 

One interesting alternative solution might be to do org-id lookups in CoVE as an Additional Check and alert people when their `id` and `name` fields don't match in the fundingOrganisation object (which would, of course, improve the data for everyone, if acted on!) 

>  it's worth thinking through what it would mean to provide a name in search results but different one in the grant record

Absolutely: this could be quite disorienting in cases where there's more substantial differences than just an ampersand. 

One way to deal with that might be:
* Restyle the top section into a summary card, using augmented data
* Make the table below contain all the fields, and be a faithful representation of the grant record itself. 
* Explain potential differences in the help site

**Comment by mariongalley** [comment]
> One interesting alternative solution might be to do org-id lookups in CoVE as an Additional Check and alert people when their `id `and `name `fields don't match in the fundingOrganisation object (which would, of course, improve the data for everyone, if acted on!)
That sounds really great - to be considered in the Data Quality Tool sprint.

> - Restyle the top section into a summary card, using augmented data
> - Make the table below contain all the fields, and be a faithful representation of the grant record itself.
Do you mean on the full grant page rather than the results card?

**Comment by mariongalley** [comment]
I think this was fixed by org pages
